lib/errors.rs

```diff
@@ -0,0 +1,83 @@
+//! Utility types and functions for errors
+
+use fatality::{Fatality, Split};
+use thiserror::Error;
+
+/// Display an error with causes.
+/// This is useful for displaying errors without converting to
+/// `miette::Report` or `anyhow::Error` first
+pub struct ErrorChain<'a>(&'a (dyn std::error::Error));
+
+impl<'a> ErrorChain<'a> {
+    pub fn new<E>(err: &'a E) -> Self
+    where
+        E: std::error::Error,
+    {
+        Self(err)
+    }
+}
+
+impl std::fmt::Display for ErrorChain<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&self.0, f)?;
+        let mut source: Option<&dyn std::error::Error> = self.0.source();
+        while let Some(cause) = source {
+            std::fmt::Display::fmt(": ", f)?;
+            std::fmt::Display::fmt(cause, f)?;
+            source = cause.source();
+        }
+        Ok(())
+    }
+}
```

Comment:
Could you add a unit test for this? We really don't have much in the way of those, and the small parts of our codebase that is easily unit testable really should be

Reply:
A benefit of unit tests for things like this is that it also serves as an example of what usage and results look like